### ISSUE TITLE
Better error messages when mutations fail due to missing return fields

### ISSUE THREAD

**Issue by skosch** [apollographql/apollo-feature-requests#65]
**I have struggled a few times now with the following scenario, and would like to request better error handling:**

* Some query Q requests some objects with fields and sub-objects
* Some mutation M runs successfully, creating more data, and returning some of Q's original objects extended by the created data
* Q breaks silently and displays nothing (sometimes there's a useless error message, not always, I haven't figured out why)

The reason Q fails is that the fields requested by M missed some of the original fields in Q. This is a problem, because the newly created data now doesn't include those fields, but query Q requires them.

Here's an example (see line 83 in App.js): https://codesandbox.io/s/wwn53np9nl

It would be great if a descriptive error message were given in this case, like `Query Q cannot be updated: Query Q requires fields A, B, C to be updated, but these fields are not included in mutation M`.

Thanks :)

**Comment by slorber** [comment]
Agreed it could be useful on mutation to warn that the mutation does not include all fields of an active query.

But please take into consideration also that sometimes we don't want to add an element to a list but simply update a single attribute of a complex cache item and providing the exhaustive list of fields to return in the mutation is annoying for such case. Apollo does not really know how you want to perform your cache update, particularly if you have a custom update function.

See https://github.com/apollographql/apollo-feature-requests/issues/72 for context of my message
